### PR TITLE
Support ConvTranspose2d

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit supports ConvTranspose2d op.

TICO-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

**Review by dayo09 — APPROVED** [review]
LGTM

File: tico/passes/convert_conv_to_conv_transpose2d.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+from tico.utils.trace_decorators import trace_graph_diff_on_pass
+from tico.utils.utils import is_target_node
+from tico.utils.validate_args_kwargs import ConvArgs
+
+
+@trace_graph_diff_on_pass
+class ConvertConvToConvTranspose2D(PassBase):
+    """
+    Convert 2-D transposed convolutions that were exported as the generic
+    `aten.convolution.default` into `aten.conv_transpose2d.input` calls.
```
Comment (jinevening) [inline tico/passes/convert_conv_to_conv_transpose2d.py:31]: Have you tested we can keep `aten.conv_transpose2d.input` if we add it to `_preserve_ops` in `run_decompositions`?

Comment (mhs4670go) [reply tico/passes/convert_conv_to_conv_transpose2d.py:31]: @jinevening Ah, yeah. I didn't know that. Let's just set it a preserved node.

File: tico/serialize/operators/op_transpose_conv.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            # luci-interpreter can't run no bias conv. Let's add zero vector for bias.
+            assert len(weight_shape) == 4
+            out_channel = weight_shape[0]
+            bias = [0.0] * out_channel  # type: ignore[assignment]
+
+        # First arguemnt is output shape of tconv.
+        tconv_output = torch.tensor(
+            [input_shape[0], output_shape[1], output_shape[2], weight_shape[0]],
+            dtype=torch.int32,
+        )
```
[the remark this answers is not in the document]
Comment (mhs4670go) [reply tico/serialize/operators/op_transpose_conv.py:148]: I'll apply this change and just make current codes assertions instead.

File: tico/serialize/operators/op_transpose_conv.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        assert len(input_shape) == 4, len(input_shape)
+        output_shape = extract_shape(node)
+        assert len(output_shape) == 4, len(output_shape)
+
+        conv_input: torch.fx.Node | circle.Tensor.TensorT = input_
+        weight_shape = list(extract_shape(weight))
+
+        if is_valid_padding(padding):
+            tconv2d_padding_type = VALID
+        elif is_same_padding(padding, input_shape, output_shape) and stride == [1, 1]:
```
[the remark this answers is not in the document]
Comment (jinevening) [reply tico/serialize/operators/op_transpose_conv.py:96]: same padding can have stride > 1.

https://medium.com/data-science/understand-transposed-convolutions-and-build-your-own-transposed-convolution-layer-from-scratch-4f5d97b2967

> “same”: the output shape is the input shape divided by the stride (convolutions) or multiplied by the stride (transposed convolutions). When the stride is equal to 1, the output shape is always the same as the input shape.

**Review by jinevening — APPROVED** [review]
LGTM

PTAL https://github.com/Samsung/TICO/pull/181#discussion_r2174992859

File: tico/serialize/operators/op_transpose_conv.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mhs4670go) [reply tico/serialize/operators/op_transpose_conv.py:96]: Maybe there's some confusion. What I'm trying to say is same with "When the stride is equal to 1, the output shape is always the same as the input shape". Before this PR, it only checked if input shape and output shape were same. But, it should have checked stride is 1 as well.
